### PR TITLE
Add PHP 8.4 to workflows

### PR DESCRIPTION
**Pull request by andypost**
Follow-up to https://github.com/consolidation/config/pull/64#issuecomment-2246242620

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -115,6 +121,10 @@ jobs:
           extensions: ${{ env.PHP_EXTENSIONS }}
           ini-values: ${{ matrix.php-ini-values }}
 
+      - name: Update composer to snapshot
+        if: matrix.php-version == '8.4'
+        run: composer self-update --snapshot
```
Comment (andypost) [inline .github/workflows/ci.yml]: Until 2.7.8 is out 8.4 should use `snapshot` https://github.com/jsonrainbow/json-schema/pull/717#issuecomment-2212513129

Comment (andypost) [reply .github/workflows/ci.yml]: Somehow `tools: composer:snapshot` does not work

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -102,6 +102,12 @@ jobs:
             dependencies: highest
             codecov: true
             php-ini-values: assert.exception=1, zend.assertions=1, opcache.enable=1, opcache.enable_cli=1, opcache.optimization_level=-1, opcache.jit_buffer_size=4096M, opcache.jit=1205
+          - os: ubuntu-latest
+            php-version: "8.4"
+            tools: composer:snapshot
+            dependencies: highest
+            codecov: true
+            php-ini-values: zend.assertions=1, opcache.enable=1, opcache.enable_cli=1, opcache.optimization_level=-1, opcache.jit_buffer_size=4096M, opcache.jit=1205
```
Comment (andypost) [inline .github/workflows/ci.yml]: JIT settings got changes https://php.watch/versions/8.4/opcache-jit-ini-default-changes

But looks `pcov` is not [updated](https://github.com/krakjoe/pcov/pull/111) or `codecov` is not compatible with JIT as there's in logs

```
Run composer self-update --snapshot
PHP Warning:  JIT is incompatible with third party extensions that override zend_execute_ex(). JIT disabled. in Unknown on line 0
```

Comment (greg-1-anderson) [reply .github/workflows/ci.yml]: We don't need to run code coverage on PHP 8.4 right now, if that helps. I don't think it's necessary to test with JIT either.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (andypost) [reply .github/workflows/ci.yml]: Fixed by passing value from matrix

File: .github/workflows/ci.yml (hunk 2 above)
Comment (andypost) [reply .github/workflows/ci.yml]: Thank you, it was pcov which ruined JIT, also adjusted variables passed to job!

**Comment by andypost** [comment]
> PHP Warning:  JIT on x86_64 doesn't support opcache.jit_buffer_size above 2G. in Unknown on line 0

new limit

**Review by greg-1-anderson — APPROVED** [review]
Nice.

**Pull request merged**